### PR TITLE
initial workflow and mini test running

### PR DESCRIPTION
**Pull request by subwaystation**
## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated
- [ ] If necessary, also make a PR on the [nf-core/pangenome branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/pangenome)

- `main.nf`: Copied the DSL2 pggb workflow from https://github.com/pangenome/pggb/pull/20 into this one. Currently we are ignoring the `nf-core` way to handle input parameters, software versions, etc. This will change in the future.
- `nextflow.config`: Set default container to `ghcr.io/pangenome/pggb:latest`
- `README.md`: Describe how developers can test the current state of the pipeline.
- `test.config`: Point to valid test data and update the default container.

I am opening this as a draft in order to compare with https://github.com/nf-core/pangenome/pull/2/files and to discuss a way forward with @heuermh .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,24 +20,32 @@ The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool
 
 ## Quick Start
 
+ **Warning:** We are currently relaying on an external docker image ghcr.io/pangenome/pggb:latest and the Dockerfile action is not running, yet. Therefore, make sure you always have the latest image. Another caveat is that you need to clone the repository before you can execute the pipeline. Once we have an automated docker image build on `nf-core`, these inconvenience will be gone.
```
Comment (Zethson) [inline README.md:23]: > relaying

bro pls

File: README.md [hunk 2]
```diff
@@ -20,24 +20,32 @@ The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool
...
     > Please check [nf-core/configs](https://github.com/nf-core/configs#documentation) to see if a custom config file to run nf-core pipelines already exists for your Institute. If so, you can simply use `-profile <institute>` in your command. This will enable either `docker` or `singularity` and set the appropriate execution settings for your local compute environment.
 
 4. Start running your own analysis!
 
     <!-- TODO nf-core: Update the example "typical command" below used to run the pipeline -->
 
     ```bash
-    nextflow run nf-core/pangenome -profile <docker/singularity/conda/institute> --input samplesheet.csv --genome GRCh37
+    docker pull ghcr.io/pangenome/pggb:latest
+    nextflow run ./main.nf -profile docker --input_fasta ~/git/test-datasets/data/DRB1-3123/DRB1-3123.fa.gz
```
Comment (Zethson) [inline README.md:48]: Ideally you would write the test profile now. 

You prepared them for a reason :)

File: conf/test.config [hunk 3]
```diff
@@ -22,4 +22,5 @@ params {
 
   // Genome references
   fasta = 'https://raw.githubusercontent.com/nf-core/test-datasets/viralrecon/genome/MN908947.3/GCA_009858895.3_ASM985889v3_genomic.200409.fna.gz'
+  input_fasta = 'https://github.com/nf-core/test-datasets/raw/pangenome/data/DRB1-3123/DRB1-3123.fa.gz'
```
Comment (Zethson) [inline conf/test.config:25]: must be raw right

File: main.nf [hunk 4]
```diff
@@ -11,27 +11,11 @@
...
-* TODO integrate this later....
-*/
-
-/*
 params.input_fasta="${baseDir}/**.fa.gz"
-params.map_pct_id=false
-params.align_pct_id=false
-params.n_secondary=false
-params.segment_length=false
+params.map_pct_id=90//false
```
Comment (Zethson) [inline main.nf:15]: Should eventually move to nextflow.config, but fine if we ignore it here

I can do that tomorrow maybe

Comment (subwaystation) [reply main.nf:15]: Yes, as stated above, this is still a TODO.

**Comment by heuermh** [comment]
Note pull requests should come from your fork of this repository against the `dev` branch, rather from branches on nf-core/pangenome itself

File: conf/test.config (hunk 3 above)
Comment (subwaystation) [reply conf/test.config:25]: Yes, it is raw. Or what do you want to tell me?

File: README.md (hunk 2 above)
Comment (subwaystation) [reply README.md:48]: There is a alreay working test profile. I can't follow what you want.

File: README.md (hunk 1 above)
Comment (subwaystation) [reply README.md:23]: Let me fix this.

**Comment by subwaystation** [comment]
> Note pull requests should come from your fork of this repository against the `dev` branch, rather from branches on nf-core/pangenome itself

Good to know. Can we still leave this or shall I redo the PR?

**Comment by heuermh** [comment]
> Can we still leave this or shall I redo the PR?

No, it is fine for now, I just went looking for the branch on your repo and couldn't find it 😉

File: README.md (hunk 1 above)
Comment (subwaystation) [reply README.md]: Done.

**Comment by subwaystation** [comment]
I will do it properly next time :+1:

File: README.md (hunk 2 above)
Comment (subwaystation) [reply README.md:48]: Did you mean to also add `nextflow run nf-core/pangenome -profile test,docker -r dev_sh` as an example on how to run the test without having to clone the repository? @Zethson

File: conf/test.config (hunk 3 above)
Comment (Zethson) [reply conf/test.config:25]: Oh I got confused that the URL did not start with raw like above. If both versions work feel free to ignore me.

File: README.md (hunk 2 above)
Comment (Zethson) [reply README.md:48]: Yes. Minus the explicit revision to your branch.

Comment (subwaystation) [reply README.md]: Done.

**Comment by subwaystation** [comment]
@Zethson This is ready from my side. Could you please take another look and merge if you are happy? Thanks!

File: README.md [hunk 5]
```diff
@@ -45,7 +43,7 @@ The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool
 
     ```bash
     docker pull ghcr.io/pangenome/pggb:latest
-    nextflow run ./main.nf -profile docker --input_fasta ~/git/test-datasets/data/DRB1-3123/DRB1-3123.fa.gz
+    nextflow run nf-core/pangenome -profile docker --input_fasta ~/git/test-datasets/data/DRB1-3123/DRB1-3123.fa.gz
```
Comment (Zethson) [inline README.md:46]: `~/git/test-datasets/data/DRB1-3123/DRB1-3123.fa.gz`

This is your local path, right?

Comment (subwaystation) [reply README.md:46]: Shall I point to the test data again or to an arbitrary path?

Comment (subwaystation) [reply README.md:46]: Yes, you are right.

Comment (Zethson) [reply README.md:46]: Test data again, please.

Comment (Zethson) [reply README.md:46]: According to nf-core standards you just write something like

`--input_fasta myfasta.fa.gz`

Note that we should use --input and not custom input parameters

Comment (subwaystation) [reply README.md]: Done.

**Review by Zethson — APPROVED** [review]
LGTM

**Pull request merged**